tensorflow/c/kernels_test.cc

```diff
@@ -466,4 +451,121 @@ TEST_F(DeviceKernelOpTest, TestAllocateOutputSize2x3) {
   EXPECT_EQ("Tensor<type: float shape: [2,3] values: [1 2 3][4 5 6]>",
             output->DebugString(100));
 }
-}  // namespace tensorflow
+
+REGISTER_OP("AllocateTempOp1").Output("output1: float");
+
+TEST_F(DeviceKernelOpTest, TestAllocateTempSizeOne) {
+  auto my_compute_func = [](void* kernel, TF_OpKernelContext* ctx) {
+    // Allocate scalar TF_Tensor
+    TF_Status* s = TF_NewStatus();
+    int64_t dim = 1;
+    TF_AllocatorAttributes alloc_attrs; 
+    alloc_attrs.struct_size = TF_ALLOCATOR_ATTRIBUTES_STRUCT_SIZE; 
+    alloc_attrs.on_host = 1; 
```

Comment:
You were asking me about how to test if it is not on host. Did that test not work eventually? I only see tests with `on_host = 1`.

If you want to write a test without host, you can do it as follows:
* validate_tensor the way you do here on CPU
* If we are running with GPU, first copy from device to host. Then, call validate_tensor.

Reply:
Just added the test, thanks Anna!